addons/advanced_fatigue/functions/fnc_calculateBodyTemperature.sqf

```diff
@@ -0,0 +1,73 @@
+/*
+* Author: Pasrules
+* Calculates body temperature in degC, normal = 37
+*
+* Arguments:
+* 0: bodyTemperature <NUMBER>
+*
+* Return Value:
+* bodyTemperature <NUMBER>
+*
+*
+* Example
+* [37] call ace_advanced_fatigue_fnc_calculateBodyTemperature
+*/
+
+#include "script_component.hpp"
+params ["_bodyTemp"];
+
+if (["ACE_Weather"] call EFUNC(common,isModLoaded)) then {
+
+    //Weather
+    private _playerAltitude = (getPosASL ACE_player) select 2;
+    private _temperature = _playerAltitude call EFUNC(weather,calculateTemperatureAtHeight);
+    private _humidity = humidity;
+    private _barometricPressure = _playerAltitude call EFUNC(weather,calculateBarometricPressure);
+    private _airDensity = [_temperature, _barometricPressure, _humidity] call EFUNC(weather,calculateAirDensity);
+    private _windSpeed = [eyePos ACE_player, false, false, false] call EFUNC(weather,calculateWindSpeed);
+    private _windFactor = linearConversion [0, 20, _windSpeed, 0, 1, true];
+
+    //Average male [kg, cm, degC]
+    private _male = [87,177,37];
```

Comment:
add these as defines in the script component file, so if these need to be changed/updated it can be done it one place?

edit: also this is kind of bad practice to define an array, only to do select operation below. Using defines is a nice way to give a number/string a meaning (`#define BODY_WEIGHT 87`) without the cost of sending the operation through the SQF engine.